lib/components/Dropdown.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { Component, PropTypes } from 'react';
+import InputLabel from './InputLabel';
+
+/**
+ * Class representing a dropdown
+ *
+ * This component expects that its children will be <options> elements
+ *
+ * Required props:
+ * - label: string. Sets the text for the input's label
+ * - id: string. Sets the select's id attribute nd the label's for attribute
+ * - children: node. Series of <option> elements
+ *
+ * Optional props:
+ * - value: string. Sets default select choice. Must match the value of one of the <option> elements. Or if an empty string, a blank placeholder is added.
+ * - required: bool. Adds required label, required attribute and aria-required='true'
+ * - errorMessage: string. If present triggers the error state and displays the error message
+ */
+export default class Dropdown extends Component {
+  /**
+   * Set initial state
+   * 
+   * hasError: tracks if the field has an error
+   * errorMessage: displayed message when the field hasError
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.value,
+      hasError: false,
+      errorMessageBody: null
+    };
+  }
+  
+  /**
+   * If a errorMessage is passed as props, adjust the state accordingly
+   */
+  componentWillMount() {
+    if (this.props.errorMessage) {
+      this.setState({
+        hasError: true,
+        errorMessageBody: this.props.errorMessage
+      });
+    }
+  }
+  
+  componentDidMount() {
```

Comment:
My original code was silly ... should have just done those state setting actions in the constructor function. 

Excellent point about the updating. There is no `didReceiveProps` that I can find, so I'm going with `componentWillUpdate` since that will cover both props and state changes.